Bottom: 2f95dab57b18d062043ebc4da91a2c2b51f087ec
Top:    2b485f17d72d951a50a5ec93c40edbddc121fa0b
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-11 16:55:05 -0500

Add role to synchronize a directory with /tmp/bacula-restores

This is used to have a secondary backup file storage because
typical restoration deletes the backup files afterward.


---
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 2d82aa67..389e047b 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -8,6 +8,10 @@
   become_user: root
   become_method: sudo
   roles:
+  - name: Create bacula-restores symlinks
+    role: bacula-restores
+    tags:
+    - bacula_restores
   - role: mariadb
     mariadb_open_firewall: true
     mariadb_remote_root_access: false
diff --git a/roles/bacula-restores/defaults/main.yml b/roles/bacula-restores/defaults/main.yml
new file mode 100644
index 00000000..b72968f6
--- /dev/null
+++ b/roles/bacula-restores/defaults/main.yml
@@ -0,0 +1,3 @@
+---
+
+bacula_storage_path: "/mnt/bacula-storage"
diff --git a/roles/bacula-restores/tasks/main.yml b/roles/bacula-restores/tasks/main.yml
new file mode 100644
index 00000000..1e614c88
--- /dev/null
+++ b/roles/bacula-restores/tasks/main.yml
@@ -0,0 +1,11 @@
+---
+# tasks file for bacula-restores
+
+- name: Synchronize bacula storage to restore directory
+  synchronize:
+    dest: "{{bacula_dest}}/" # required. Path on the destination host that will be synchronized from the source; The path can be absolute or relative.
+    src: "{{bacula_storage_path}}/" # required. Path on the source host that will be synchronized to the destination; The path can be absolute or relative.
+    delete: yes # not required. Delete files in C(dest) that don't exist (after transfer, not before) in the C(src) path. This option requires C(recursive=yes).
+    archive: yes # not required. Mirrors the rsync archive flag, enables recursive, links, perms, times, owner, group flags and -D.
+    link_dest: "{{bacula_storage_path}}/" # not required. add a destination to hard link against during the rsync.
+  delegate_to: "{{ inventory_hostname }}"
